### PR TITLE
[rest-server] in-cluster config load in a pod within rbac enabled k8s

### PR DESCRIPTION
**Pull request by ydye**

### CODE REVIEW

File: src/rest-server/src/config/kubernetes.js [hunk 1]
```diff
@@ -27,15 +28,38 @@ const {
...
+  apiserverConfig.uri = K8S_APISERVER_URI ? K8S_APISERVER_URI : `https://${host}:${port}`;
+
+  try {
+    // Will be a buffer since SSL context can receive a buffer.
+    apiserverConfig.ca = readFileSync(caPath);
+    // Will be a string since http header can only receive a string.
+    apiserverConfig.token = readFileSync(tokenPath, 'ascii');
+  } catch (error) {
+    // eslint-disable-next-line no-console
+    console.log('failed to init rbac config. Please check your clusters\' config');
```
Comment (abuccts) [inline src/rest-server/src/config/kubernetes.js]: please do not use `console.log`

Comment (ydye) [reply src/rest-server/src/config/kubernetes.js]: alternative solution thanks.

File: src/rest-server/src/config/kubernetes.js [hunk 2]
```diff
@@ -27,15 +29,38 @@ const {
...
+  apiserverConfig.uri = K8S_APISERVER_URI ? K8S_APISERVER_URI : `https://${host}:${port}`;
+
+  try {
+    // Will be a buffer since SSL context can receive a buffer.
+    apiserverConfig.ca = readFileSync(caPath);
+    // Will be a string since http header can only receive a string.
+    apiserverConfig.token = readFileSync(tokenPath, 'ascii');
+  } catch (error) {
+    // eslint-disable-next-line no-console
+    logger.info('failed to init rbac config. Please check your clusters\' config');
```
[the remark this answers is not in the document]
Comment (ydye) [reply src/rest-server/src/config/kubernetes.js]: DOne

File: src/rest-server/src/config/kubernetes.js [hunk 3]
```diff
@@ -27,15 +29,38 @@ const {
...
+    process.exit(1);
```
[the remark this answers is not in the document]
Comment (ydye) [reply src/rest-server/src/config/kubernetes.js]: DoNe

File: src/rest-server/deploy/rest-server.yaml.template [hunk 4]
```diff
@@ -106,11 +106,12 @@ spec:
         - name: GITHUB_PATH
           value: {{ cluster_cfg['rest-server']['github-path'] }}
 {% endif %}
-{% if cluster_cfg['layout']['kubernetes']['api-servers-url'].startswith('https://') %}
-        - name: K8S_APISERVER_CA_FILE
```
Comment (abuccts) [inline src/rest-server/deploy/rest-server.yaml.template]: `K8S_APISERVER_CA_FILE` is still used in the code, do you need to delete it here?

Comment (ydye) [reply src/rest-server/deploy/rest-server.yaml.template]: Yes, the env is kept in non-rbac mode for expandability in the future. And in rbac enabled cluster, thoses two values are always a hardcode value, so just remove them from deployment code at this stage.

**Pull request merged**